networking/eth2/src/main/java/tech/pegasys/teku/networking/eth2/gossip/topics/GossipSubValidationUtil.java

```diff
@@ -0,0 +1,34 @@
+/*
+ * Copyright 2020 ConsenSys AG.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package tech.pegasys.teku.networking.eth2.gossip.topics;
+
+import io.libp2p.core.pubsub.ValidationResult;
+import tech.pegasys.teku.statetransition.validation.InternalValidationResult;
+
+public class GossipSubValidationUtil {
+
+  public static ValidationResult fromInternalValidationResult(InternalValidationResult result) {
+    switch (result) {
+      case ACCEPT:
+        return ValidationResult.Valid;
+      case SAVE_FOR_FUTURE:
+      case IGNORE:
+        return ValidationResult.Ignore;
+      case REJECT:
+        return ValidationResult.Invalid;
+      default:
+        throw new UnsupportedOperationException("Unexpected validation result: " + result);
```

Comment:
Probably should be an `IllegalArgumentException`.

Reply:
Donne.